### PR TITLE
Added auto-labelling to PRs

### LINKED ISSUE #1236: Danger: Add wip label for PRs that doesn't comply
I think it's a great idea for danger to add a `wip` label for PRs that doesn't comply with the guideline and won't check it unless the `wip` label is removed. If it complies with the guideline, add a `pending review` label.

### PR DESCRIPTION
**Pull request by anirudhprabhakaran3**
Fixes #1236 

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [X] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [X] code is in uniquely-named feature branch and has no merge conflicts 📁
* [X] screenshots/GIFs are attached 📎 in case of UI updation
* [X] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

Thanks!

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -2,6 +2,15 @@ name: tests
...
 jobs:
 
+  add-label:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: buildsville/add-remove-label@v1
+      with:
+        token: ${{ secrets.GITHUB_TOKEN }}
+        label: work-in-progress
+        tyle: add
```
Comment (TildaDares) [inline .github/workflows/tests.yml:12]: @anirudhprabhakaran3 Shouldn't tyle be type?

Comment (anirudhprabhakaran3) [reply .github/workflows/tests.yml]: Yeah, I just fixed that in the newest commit. Thanks for pointing it out!

**Comment by cesswairimu** [comment]
looking good @anirudhprabhakaran3 , did you find a way to ignore the code climate checks? . thanks

**Comment by anirudhprabhakaran3** [comment]
Aah yes, I added the labelling to the core tests itself, so it completely ignores the code climate tests.
So now, the workflow looks like this:
- Adding the `work-in-progress` label, unit tests, system tests, integration tests and functional tests.
- If, and only if, all these jobs run successfully, the old label is removed and the `ready-for-review` label is attached

**Comment by cesswairimu** [comment]
Fantastoic :tada: :tada:  I will wait for a confirmation from Ruth and Jeff then we are ready to merge. Thanks again

**Review by cesswairimu — APPROVED** [review]
:tada: :tada:

**Review by RuthNjeri — APPROVED** [review]
Looks good to me, thanks @anirudhprabhakaran3 🎉 
Thanks for the reviews @TildaDares and @cesswairimu ❤️

**Review by jywarren — APPROVED** [review]

**Comment by jywarren** [comment]
Wow, very cool! Love it!

**Pull request merged**